compiler_opt/distributed/local/buffered_scheduler_test.py

```diff
@@ -91,44 +91,44 @@ def task():
     # should've been assigned elsewhere if load balancing works.
     self.assertEqual(call_count[0], 2)
 
-  def test_deadlock(self):
-    lock = threading.Lock()
-
-    def should_not_deadlock(_):
-      with lock:
-        pass
-      future = concurrent.futures.Future()
-      future.set_result(0)
-      return future
-
-    t = threading.Thread(target=None)
-
-    def time_able_job(_):
-      nonlocal t
-      future = concurrent.futures.Future()
-
-      def task():
-        with lock:
-          future.set_result(0)
-
-      t = threading.Thread(target=task, daemon=True)
-      return future
-
-    work = [time_able_job, should_not_deadlock]
-
-    futures = buffered_scheduler.schedule(work, [None], buffer=1)
-    # At this point, time_able_job has a done_callback to
-    # should_not_deadlock, so we can go ahead and complete the job.
-    t.start()
-    # Sleep for up to 1 second.
-    for _ in range(10):
-      if futures[0].done() and futures[1].done():
-        break
-      time.sleep(0.1)
-    # If >1s, and it's still not done, it's probably deadlocked.
-    self.assertTrue(futures[0].done())
-    self.assertTrue(futures[1].done())
-    t.join()
+  # def test_deadlock(self):
```

Comment:
because we shouldn't check in commented code (as a matter of principle), how about peeling this in a gist and referencing that in the comment?

Also, could you please add a comment on LocalWorkerPool about the interaction w the scheduler - we can then move those comments to a common abstraction later, when that appears.

Reply:
(lgtm otherwise)